### DISCUSSION TITLE
Python中read()、readline()和readlines()三者間的區別和用法

### DISCUSSION THREAD

**Discussion by 107602505** [1101-GS4719-C/Python-Community-GS4719-C#254] [General]
之前有點不太清楚這三個指令的區別，上網查後整理了一下大家可以參考看看~
read()方法：
1、讀取整個檔案，將檔案內容放到一個字串變數中 (str)
2、如果檔案大於可用記憶體，不可能使用這種處理
![image](https://user-images.githubusercontent.com/90819282/148092723-83ab8b51-fc40-4b87-ba68-d10508349b48.png)

readline()方法：
1、readline()每次讀取一行，比readlines()慢得多
2、readline()返回的是一個字串物件(str)，儲存當前行的內容
![image](https://user-images.githubusercontent.com/90819282/148093720-3769be63-6feb-4c88-85f9-88280b71f101.png)

readlines()方法：
1、一次性讀取整個檔案
2、自動將檔案內容分析成一個行的列表(list)
![image](https://user-images.githubusercontent.com/90819282/148093493-a4c9c3e5-ea7d-44f2-8b51-68525a79f51c.png)

**Comment by MAKERRRofKEVIN** [comment]
感謝同學分享，超實用!!!!上次小考看到這三條我一臉矇逼不知道要怎麼選，看完你的解釋之後都弄懂了

**Comment by chupochun** [comment]
謝謝你的分享!!我了解這三者的差異了

**Comment by ghost** [comment]
之後有用到這塊會過來參考一下，辛苦了

**Comment by pekoranoinu** [comment]
感謝分享讀檔寫檔的各種方式。

**Comment by xkllkx** [comment]
謝謝樓主分享，最近專題有用到讀取txt的程式，剛好解決了我的疑問。
